genesyscloud/routing_queue/genesyscloud_routing_queue_proxy.go

```diff
@@ -116,49 +179,95 @@ func getAllRoutingQueuesFn(ctx context.Context, p *RoutingQueueProxy) (*[]platfo
 	return &allQueues, resp, nil
 }
 
+func createRoutingQueueFn(ctx context.Context, p *RoutingQueueProxy, createReq *platformclientv2.Createqueuerequest) (*platformclientv2.Queue, *platformclientv2.APIResponse, error) {
+	return p.routingApi.PostRoutingQueues(*createReq)
+}
+
 // getRoutingQueueByIdFn is the implementation for retrieving a routing queues in Genesys Cloud
 func getRoutingQueueByIdFn(ctx context.Context, p *RoutingQueueProxy, queueId string) (*platformclientv2.Queue, *platformclientv2.APIResponse, error) {
 	queue := rc.GetCacheItem(p.RoutingQueueCache, queueId)
 	if queue != nil {
 		return queue, nil, nil
 	}
 
-	queue, resp, err := p.routingApi.GetRoutingQueue(queueId)
+	return p.routingApi.GetRoutingQueue(queueId)
+}
+
+func getRoutingQueueByNameFn(ctx context.Context, p *RoutingQueueProxy, name string) (string, *platformclientv2.APIResponse, bool, error) {
+	queues, resp, err := GetAllRoutingQueuesFn(ctx, p, name)
 	if err != nil {
-		return nil, resp, fmt.Errorf("failed to retrieve routing queue by id %s: %s", queueId, err)
+		return "", resp, false, err
 	}
 
-	return queue, resp, nil
+	if queues == nil || len(*queues) == 0 {
+		return "", resp, true, fmt.Errorf("no routing queue found with name %s", name)
+	}
+
+	for _, queue := range *queues {
+		if *queue.Name == name {
```

Comment:
You'll probably need to use `normalizeQueueName` here like it was in the original